File: src/wfrp-helper-web/src/model/creature/PersonalRangedWeapon.js
import Ammunition   from "../crafting/Ammunition";
import RangedWeapon from "../crafting/RangedWeapon";
import Entity       from "../Entity";

export default class PersonalRangedWeapon extends Entity {
    rangedWeapon: RangedWeapon;
    ammunition: Ammunition;
    ammunitionAmount: number = 0;


    updateWith(entity: PersonalRangedWeapon): PersonalRangedWeapon {
        super.updateWith(entity);
        super.updateEntityProp(entity, 'rangedWeapon', () => new RangedWeapon());
        super.updateEntityProp(entity, 'ammunition', () => new Ammunition());
        return this;
    }
}

File: src/main/java/pl/khuzzuk/wfrp/helper/model/creature/PersonalRangedWeapon.java
package pl.khuzzuk.wfrp.helper.model.creature;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.envers.Audited;
import org.hibernate.envers.RelationTargetAuditMode;
import pl.javahello.DTO;
import pl.khuzzuk.wfrp.helper.model.crafting.inventory.Ammunition;
import pl.khuzzuk.wfrp.helper.model.crafting.inventory.RangedWeapon;
import pl.khuzzuk.wfrp.helper.repo.BaseEntity;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Getter
@Setter
@Entity
@Table(schema = "creature", name = "person_ranged_weapons")
@Audited
@DTO
public class PersonalRangedWeapon extends BaseEntity {
    @ManyToOne
    @JoinColumn(name = "ranged_weapon_id")
    @Audited(targetAuditMode = RelationTargetAuditMode.NOT_AUDITED)
    private RangedWeapon rangedWeapon;
    @ManyToOne
    @JoinColumn(name = "ammunition_id")
    @Audited(targetAuditMode = RelationTargetAuditMode.NOT_AUDITED)
    private Ammunition ammunition;
    private int ammunitionAmount;
}
